src/requestManager.cpp

```diff
@@ -587,6 +667,31 @@ void CRequestManager::SendRequests()
             }
         }
     }
+    // send batched requests if any.
+    if (fBatchBlockRequests && !mapBatchBlockRequests.empty())
+    {
+        for (auto iter : mapBatchBlockRequests)
+        {
+            LEAVE_CRITICAL_SECTION(cs_objDownloader);
```

Comment:
we should be able to put this and the ENTER outside of the for loop

Reply:
thanks, ok done...also moved up cs_main...we only need to take cs_main once as well.